Messaging.md

```diff
@@ -363,92 +371,182 @@ Witnesses should be sorted by the `input_serial_id` sent in `funding_input` defi
+   * [`num_outcomes*string`:`outcomes`]
 
 This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
 
-Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+Note that `outcomes[i]` is the outcome value itself and not its hash that will be signed by the oracle.
 
 #### `digit_decomposition_event_descriptor`
 
 1. implements: `event_descriptor`
-1. type: 55306
+1. type: 1
 1. data:
    * [`bigsize`:`base`]
    * [`bool`:`is_signed`]
    * [`string`:`unit`]
    * [`int32`:`precision`]
    * [`u16`:`nb_digits`]
 
+### The `oracle_event_timestamp` Type
+
+This type contains timestamp information about when an `oracle_event` is to happen.
+
+There are two kinds of `oracle_event_timestamps`: fixed timestamps for events with a
+known (expected) epoch time at which the event will occur, and range for events whose
+time is not known ahead of time and also for events which may occur at any time in the
+specified range.
+
+When using an `oracle_event_timestamp_range` for events which may occur at any time,
+the `earliest_expected_time_epoch` should correspond to the time at which the oracle
+announcement was created and `latest_expected_time_epoch` should correspond to the
+latest time for which the oracle will be responsible for attesting to the event's occurence.
+Note that it is possible to have functionally open-ended events through the use of very large
+`latest_expected_time_epoch` timestamps.
+
+#### `fixed_oracle_event_timestamp`
+
+1. implements: `oracle_event_timestamp`
+1. type: 0
+1. data:
+   * [`u32`:`expected_time_epoch`]
+
+#### `range_oracle_event_timestamp`
+
+1. implements: `oracle_event_timestamp`
+1. type: 1
+1. data:
+   * [`u32`:`earliest_expected_time_epoch`]
+   * [`u32`:`latest_expected_time_epoch`]
+
 ### The `oracle_event` Type
 
 This type contains information provided by an oracle on an event that it will attest to.
 See [the Oracle specifications](./Oracle.md#oracle-event) for more details.
 
-**For backward compatibility reasons, this type is currently serialized as a TLV and uses u16 for collection prefix. It is expected to be changed to the new serialization format in a near future.**
-
 #### `oracle_event`
 
-1. type: 55330
 1. data:
-   * [`u16`:`nb_nonces`]
-   * [`nb_nonces*x_point`:`oracle_nonces`]
-   * [`u32`:`event_maturity_epoch`]
+   * [`oracle_event_timestamp`:`timestamp`]
    * [`event_descriptor`:`event_descriptor`]
    * [`string`:`event_id`]
 
+### The `oracle_keys` Type
+
+This type contains static oracle information and can be used to import trusted oracles into wallets.
+This information should be saved by DLC nodes.
+
+#### `oracle_keys`
+
+1. data:
+   * [`x_point`:`announcement_public_key`]
+   * [`string`:`oracle_name`]
+   * [`string`:`oracle_description`]
+   * [`u32`:`timestamp`]
+   * [`oracle_schemes`:`oracle_schemes`]
+   * [`signature`:`oracle_metadata_signature`]
```

Comment:
@nkohen why is there a signature here and then one on the announcement? Isn't it redundant?

Reply:
So as it stands this message is static and can be imported by a wallet independently of an oracle announcement right? In this case it is important that the oracle metadata is what the oracle put down. If the keys are what you expect but someone takes the message and re-writes something about the scheme or description or such, this could be bad for wallet impls

Reply:
If it is supposed to be a standalone message, does it make sense to include it in every single announcement? I'll go ahead with that for now but I feel it's also worth discussing.